### ISSUE TITLE
currency_rate_update not updating EUR currencies when base currency != EUR using ECB feed

### ISSUE THREAD

**Issue by pjmcdermott** [OCA/account-financial-tools#186]
I am using the currency_rate_update module on Odoo 8.0.20150330.

My base currency is GBP and I have additional working currencies USD and EUR. I am using the ECB update method.

For "currencies to update with this service" I have set USD and GBP. _I am unable to select a EUR option_. Now I realise that the ECB rates provided are conversion _rates_ from foreign currencies to Euros. But from my perspective as a GBP base currency user it is EUR and USD which I wish to update. The correct EUR update from my perspective is 1/GBP rate.

The USD update appears to be correct. For instance for 9/4/2015 I have 1.482389, and xe.com is telling me that the spot rate as I write is 1.47153. So currency_rate_update appears to be understanding that my base currency is GBP and making the appropriate adjustments using the EUR:USD rate and the EUR:GBP rate to get the GBP:EUR. It is just not able to do a similar adjustment to create a GBP:EUR rate.

Given that the ECB feed provides sufficient information to provide Base currency to EUR exchange rates, it should be possible to select EUR as a rate to update if the base currency != EUR.

**Comment by pjmcdermott** [comment]
Hey! I found out that the update service was coded to handle the case described above, it's just that the EUR currency was missing from <code>ECB_supported_currency_array</code> in <code>model/currency_rate_upate.py</code>. Adding "EUR" to the array allowed me to select EUR as a currency to be updated.

Please see the diff below:

``` diff
larkfield model # diff -c currency_rate_update.py.orig currency_rate_update.py 
*** currency_rate_update.py.orig    2015-04-09 23:20:20.715563413 +0100
--- currency_rate_update.py 2015-04-09 23:21:02.192690783 +0100
***************
*** 102,111 ****
      "USD", "UYU", "VEF", "VND", "ZAR"]

  ECB_supported_currency_array = [
!     "AUD", "BGN", "BRL", "CAD", "CHF", "CNY", "CZK", "DKK", "GBP", "HKD",
!     "HRK", "HUF", "IDR", "ILS", "INR", "JPY", "KRW", "LTL", "MXN", "MYR",
!     "NOK", "NZD", "PHP", "PLN", "RON", "RUB", "SEK", "SGD", "THB", "TRY",
!     "USD", "ZAR"]

  MX_BdM_supported_currency_array = [
      "ARS", "AUD", "BBD", "BMD", "BOB", "BRL", "BSD", "BZD", "CAD", "CHF",
--- 102,111 ----
      "USD", "UYU", "VEF", "VND", "ZAR"]

  ECB_supported_currency_array = [
!     "AUD", "BGN", "BRL", "CAD", "CHF", "CNY", "CZK", "DKK", "EUR", "GBP",
!     "HKD", "HRK", "HUF", "IDR", "ILS", "INR", "JPY", "KRW", "LTL", "MXN",
!     "MYR", "NOK", "NZD", "PHP", "PLN", "RON", "RUB", "SEK", "SGD", "THB",
!     "TRY", "USD", "ZAR"]

  MX_BdM_supported_currency_array = [
      "ARS", "AUD", "BBD", "BMD", "BOB", "BRL", "BSD", "BZD", "CAD", "CHF",
```

I'm afraid my github skills are not that great so I can't do a pull request...

**Comment by yvaucher** [comment]
Fixed by #187
